### DISCUSSION TITLE
Secure context and hostname config

### DISCUSSION THREAD

**Discussion by FernetB** [ionic-team/capacitor#5387] [Q&A]
Hi! I think I'm missing some concepts or something, because I do not completely understand how it works.

I know that you can provide secure context with domains under `https`, or  simply with `localhost`. But recently I changed the `hostname` for the domain that we have under https and the `scheme` to https and all work, we still can use `getUserMedia` that needs secure context and even `window.isSecureContext` equals `true`, when clearly the docs says:

![image](https://user-images.githubusercontent.com/47339424/150414505-d1b4e18c-8f54-47d9-be6e-03bf29d8aad7.png)

Someone can explain me what is happenning?

I'm using capacitor v3 and tested it with android

**Comment by jcesarmobile** [comment]
If you use http then only localhost is considered secure.
Id you use https then the host doesn’t matter, all are secure.

**Reply by FernetB** [reply]
Yes I know that, but I do not understand why the docs does not recommends to change `hostname` because of losing secure context, when you can simple change scheme to `https` and thats all

**Reply by jcesarmobile** [reply]
Because http is the default, so if you are using http and change the hostname you lose the secure context, the docs consider you are using the default values for the other config options.
But maybe a note could be added.

**Reply by FernetB** [reply]
If that's true I think that the doc needs to be changed, because all the people I saw in `set-cookie` related problems said that they can't change `hostname` because of losing secure context and even with `cors` where Ionic itself encourages you to allow localhost in the cors header. My doubt is, why is `http` the default scheme and what do you lose by chaging it to `https`
